flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -835,8 +839,15 @@ public void updateAccumulators(final AccumulatorSnapshot accumulatorSnapshot) {
                         mainThreadExecutor);
     }
 
-    private void startCheckpointScheduler(final CheckpointCoordinator checkpointCoordinator) {
+    @Override
+    public void stopCheckpointScheduler() {
+        getCheckpointCoordinator().stopCheckpointScheduler();
```

Comment:
`getCheckpointCoordinator` can return `null` if checkpointing is not enabled. Maybe add a precondition that checkpointing is enabled.